src/MySqlEventStore.php

```diff
@@ -552,11 +512,166 @@ public function stopProjection(string $name): void
         ]);
     }
 
+    public function fetchStreamNames(
+        ?string $filter,
+        bool $regex,
+        ?MetadataMatcher $metadataMatcher,
+        int $limit,
+        int $offset
+    ): array {
+        if (null === $filter && $regex) {
+            throw new Exception\InvalidArgumentException('No regex pattern given');
+        }
+
+        if ($regex && false === @preg_match("/$filter/", '')) {
+            throw new Exception\InvalidArgumentException('Invalid regex pattern given');
+        }
+
+        $where = [];
+        $values = [];
+
+        if (null !== $filter && $regex) {
+            $where[] = '`real_stream_name` REGEXP :filter ';
+            $values[':filter'] = $filter;
+        } elseif (null !== $filter && ! $regex) {
+            $where[] = '`real_stream_name` = :filter ';
+            $values[':filter'] = $filter;
```

Comment:
only exact match! should be change that to `LIKE %filter%` ?

If so, this has to be changed at InMemoryEventStore as well